### DISCUSSION TITLE
WS11WineCX20.0.4 issues with 32bit apps and AMD based Hackintoshes

### DISCUSSION THREAD

**Discussion by tomnic79** [The-Wineskin-Project/WineskinServer#187] [Q&A] [answered]
Hi, I'm Nicola and I'm trying to make AMD Hackintoshes as perfect as possible, trying to fix all kind of apps not running properly on Ryzen or Threadripper Cpus.

I came here to try to solve an issue "born" with Crossover 20/21 and AMD based Hackintoshes: we cannot run 32 bit apps reliably... For example if I try to install Steam it hangs during its self-updating... Putty works but has some problems displaying help if you click help button (blank page and app hanging).

Something similar happens with Wineskin Winery using WS11WineCX20.0.4 engine, even though sometimes randomly I can even log in into Steam but nothing is rendered inside its windows, Putty instead works perfectly even pressing help button.

A nice "progress" but still a no go. 

Can we try to solve this problem since it doesn't depend on the lack of opemu32 since Catalina? Some infos found here: 

https://github.com/AMD-OSX/bugtracker/issues/6

The extreme synopsis is...

AMD CPU is not totally unable to run ANY 32bit code in previous kernel-patched macOS. Then why when opening wine/32bit app on kernels that without OPEMU, we got illegal instrument error? That's because AMD lacks sysenter/sysexit in 64-bit compatible mode, only the two instrument which commonly used for implementing syscall in 32bit OS, any other i386 instrument working natively;

In Catalina (and higher), since no 32bit syscall exist, sysenter/sysexit was totally replaced by syscall/sysret, which already handles well on AMD CPU;

Wine32on64 does not use any other instrument that AMD does not support, or at least invalid opcode exception should be watched.

It's a big mystery... any clue?

**Comment by Gcenx** [comment, marked as answer]
First let me say I’ve never used an AMD hackintosh but have been made aware of problems on these systems that don’t happen on Intel hackintosh systems.

> Putty works but has some problems displaying help if you click help button (blank page and app hanging).

Never used this application so I’m unfamiliar with any possible bugs or required workarounds. But you say this works just fine in WineCX20.0.4 but not crossover a little strange but I usually back-port additional fixes.

> even though sometimes randomly I can even log in into Steam but nothing is rendered inside its windows

Steam has it’s own requirements and are listed in the [wiki#steam](https://github.com/Gcenx/WineskinServer/wiki#steam)

wine32on64 is really just a bunch of hacks with trunks generated using a custom llvm/clang-8. So modifying this part is a nightmare resulting in wine32on64 being much more barebones when compared to the wine64 slice.
